### DISCUSSION TITLE
UI Minimization

### DISCUSSION THREAD

**Discussion by nckrwlmn** [Ylianst/MeshCentral#6146] [Ideas]
Is anyone else wondering why almost 15 percent of the vertical space in the UI is taken up with a huge masthead and menus upon menus? Can we please have the option in the config file to remove the massive upper forehead on the webpage? It especially bad for those who have poor vision and ctrl+zoom the UI in. Makes the problem worse. Please give the option to remove the whole thing. The logout button could be in the same row as the "mydevices" button.

Also, a simple color picker or image picker for the background would also be a welcome change. Thank you Ylian.

**Comment by si458** [comment]
you can specify `hide: 1`(example) inside of your `domains: ""` in config.json to hide certain panels for all users
OR
you can add `?hide=NUMBER` in the url bar to hide per session, etc.
```
"hide": {
  "type": "integer",
  "default": 0,
  "description": "Sum of: 1 = Hide header, 2 = Hide tab, 4 = Hide footer, 8 = Hide title, 16 = Hide left bar, 32 = Hide back buttons"
},
```

**Reply by nckrwlmn** [reply]
how does this fit into the domains section?

**Reply by si458** [reply]
```
{
"domains": {
  "": {
    "hide": 1
  }
}
```

**Comment by nckrwlmn** [comment]
You guys are geniuses. How is this already in the software? Thank you.

**Reply by si458** [reply]
No worries,
In theory u can read this file https://github.com/Ylianst/MeshCentral/blob/master/meshcentral-config-schema.json as it contains all the different options for config.json, and what they do and where to place them.
You can also load up ur config.json inside visual studio code. And use the suggestion prompt and it will auto load that file and give u option's too ;)
We do know too that we need to update the docs https://ylianst.github.io/MeshCentral/

**Comment by nckrwlmn** [comment]
I took the most advanced config.json I could find and used it. But the hide feature was not there. 

One other thing: If I use windows rdp into a server and I go full screen it's almost like a native experience. What is preventing the browser based experience from being the same? I have all of the settings cranked, I have desktop multiplexing on and high quality desktop on in the config file. Why doesn't the agent remote desktop seem the same? Its just a lower quality resolution.

**Reply by si458** [reply]
I've just checked and it's there in the sample advanced json
```
"___hide__": "Sum of: 1 = Hide header, 2 = Hide tab, 4 = Hide footer, 8 = Hide title, 16 = Hide left bar, 32 = Hide back buttons",
"_hide": 4,
```

Any remote software won't feel native like ur sat in front of the computer.

But The meshagent remote option captures the frames bit by bit and encoded/decoders them bit by bit, where as the rdp option just tunnels a 3389 tcp session and displays the output, so the remote device does the encoding etc not the meshcentral server

**Reply by nckrwlmn** [reply]
So with the top header bar hidden, i cant no longer get to the fullscreen button to maximize the window. I can see its probably there (based on the two green numlock and caps lock buttons) but i just cant get to it. Can you guys please move it down when that part is hidden?

**Reply by nckrwlmn** [reply]
> I've just checked and it's there in the sample advanced json
> 
> ```
> "___hide__": "Sum of: 1 = Hide header, 2 = Hide tab, 4 = Hide footer, 8 = Hide title, 16 = Hide left bar, 32 = Hide back buttons",
> "_hide": 4,
> ```
> 
> Any remote software won't feel native like ur sat in front of the computer.
> 
> But The meshagent remote option captures the frames bit by bit and encoded/decoders them bit by bit, where as the rdp option just tunnels a 3389 tcp session and displays the output, so the remote device does the encoding etc not the meshcentral server

So if i do an rdp session in meshcentral, the tcp tunnel is going to the server and I'm viewing that, instead of directly?

Like i say, a direct rdp session feels like its my native PC. Its really good.
